src/main/java/stirling/software/SPDF/controller/api/misc/ExtractImagesController.java

```diff
@@ -63,7 +60,7 @@ public ResponseEntity<byte[]> extractImages(@ModelAttribute PDFExtractImagesRequ
             throws IOException, InterruptedException, ExecutionException {
         MultipartFile file = request.getFileInput();
         String format = request.getFormat();
-        boolean allowDuplicates = request.isAllowDuplicates();
```

Comment:
Please dont remove!

Reply:
ok,

Reply:
The multithreading approach for image extraction was revised to improve thread safety, especially after errors were encountered during testing after adding a new feature(saving PDFs to temp files. Despite these improvements, the output still shows inconsistency, with a variable number of images being extracted (between 60 and 68) from the same test PDF.

Reply:
Looking at code one issue is imageIndex needs to be atomicInteger to be thread safe

also instead of sync block maybe just ConcurrentHashSet ?

Reply:
I have included atomic integer- image Index. I will include the second one too.

Reply:
Nice! lemme know if it helps at all

Reply:
Despite the addition of atomic locks and using a ConcurrentHashSet for image extraction, the issue of inconsistent image output persists. The number of extracted images continues to vary, ranging from 65 to 68. I am not aware if it is really thread safety issue or a formatting issue
I am seeing the following output for skipped images : 14:13:57.276 [pool-3-thread-2] ERROR s.s.S.c.a.m.ExtractImagesController - Error processing image from page 2
java.io.IOException: ICCBased colorspace array must have a stream as the second element